BillTrack.Application/Services/WebApiService.cs

```diff
@@ -79,7 +79,7 @@ private async Task<T> GetByIdAsyncOrThrow<T>(IGenericRepository<T> repository, G
 
         if (entity == null)
         {
-            throw new NotFoundException($"Entity of type {typeof(T)} with ID {id} not found.");
+            throw new NotFoundException($"Entity with ID {id} not found.");
```

Comment:
it is better to have a static error message. The dynamic elements (`id`) could be passed as additional parameter to a custom exception (for example, as a JSON string).
Or even as a string with most static content:
```Entity with ID not found ({id})```
It will make it easier to group errors later.